flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/dql/SqlShowDatabases.java

```diff
@@ -29,14 +33,64 @@
 import java.util.Collections;
 import java.util.List;
 
-/** SHOW Databases sql call. */
+import static java.util.Objects.requireNonNull;
+
+/**
+ * SHOW Databases sql call. The full syntax for show databases is as followings:
+ *
+ * <pre>{@code
+ * SHOW DATABASES [ ( FROM | IN ) catalog_name] [ [NOT] (LIKE | ILIKE)
+ * <sql_like_pattern> ] statement
+ * }</pre>
+ */
 public class SqlShowDatabases extends SqlCall {
 
     public static final SqlSpecialOperator OPERATOR =
             new SqlSpecialOperator("SHOW DATABASES", SqlKind.OTHER);
 
-    public SqlShowDatabases(SqlParserPos pos) {
+    private final String preposition;
+    private final SqlIdentifier catalogName;
+    private final String likeType;
+    private final SqlCharStringLiteral likeLiteral;
+    private final boolean notLike;
+
+    public String[] getCatalog() {
+        return catalogName == null || catalogName.names.isEmpty()
+                ? new String[] {}
+                : catalogName.names.toArray(new String[0]);
+    }
+
+    public SqlShowDatabases(
+            SqlParserPos pos,
+            String preposition,
+            SqlIdentifier catalogName,
+            String likeType,
+            SqlCharStringLiteral likeLiteral,
+            boolean notLike)
+            throws ParseException {
         super(pos);
+        this.preposition = preposition;
+
+        this.catalogName =
+                preposition != null
+                        ? requireNonNull(catalogName, "Catalog name must not be null.")
+                        : null;
+        if (this.catalogName != null && this.catalogName.names.size() > 1) {
+            throw new ParseException(
+                    String.format(
+                            "Show databases from/in identifier [ %s ] format error, catalog cannot contain dot character.",
```

Comment:
As something that I'm unsure about... I wonder if a catalog name can contain a `.` if it quoted with backticks.

If so, then this statement is inaccurate.

Reply:
I believe you can quote any character.

Reply:
Given the responses, could you please update the exception to e.g.

```
"Show databases from/in identifier [ %s ] format error, catalog must be a single part identifier."
```